File: examples/modulith-clinical-tool-jpa/src/main/java/io/zenwave360/example/clinicaltool/modules/surveys/service/dtos/SurveySectionOutput.java
package io.zenwave360.example.clinicaltool.modules.surveys.service.dtos;

import io.zenwave360.example.clinicaltool.modules.surveys.domain.*;
import java.io.Serializable;
import java.math.*;
import java.time.*;
import java.util.*;
import jakarta.validation.constraints.*;

/**
 * SurveySectionOutput.
 */
@lombok.Getter @lombok.Setter
public  class SurveySectionOutput  implements Serializable {


    @NotNull @Size(max = 254)
    private String name ;

    @NotNull @Size(max = 254)
    private String title ;

    
    private String summary ;

    
    private List<QuestionTranslationOutput> questions = new ArrayList<>();






        public SurveySectionOutput addQuestions(QuestionTranslationOutput questions) {
        this.questions.add(questions);
        return this;
        }

}

File: examples/modulith-klinical-tool-jpa/src/main/kotlin/io/zenwave360/example/clinicaltool/modules/surveys/service/dtos/SurveySectionOutput.kt
package io.zenwave360.example.clinicaltool.modules.surveys.service.dtos

import io.zenwave360.example.clinicaltool.modules.surveys.domain.*
import java.io.Serializable
import java.math.*
import java.time.*
import jakarta.validation.constraints.*

/**
 * SurveySectionOutput.
 */
data class SurveySectionOutput(

    @NotNull @Size(max = 254)
    val name: String?  = null,

    @NotNull @Size(max = 254)
    val title: String?  = null,

    
    val summary: String?  = null,

    
    val questions: MutableList<QuestionTranslationOutput> = mutableListOf()

)  : Serializable {





    fun addQuestions(questions: QuestionTranslationOutput): SurveySectionOutput {
        this.questions += questions
        return this
    }

}
